src/Swoole/Dispatch/RiskyRequest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Utopia\Swoole\Dispatch;
+
+use Swoole\Server;
+
+abstract class RiskyRequest extends ContextualDispatch
+{
+    /**
+     * @param float $riskyWorkersPercent Decimal form 0 to 1
+     */
+    public function __construct(
+        int $dispatcherMapSize,
+        private readonly float $riskyWorkersPercent
+    ) {
+        parent::__construct($dispatcherMapSize);
+        if ($this->riskyWorkersPercent < 0 || $this->riskyWorkersPercent > 1) {
+            throw new \InvalidArgumentException('riskyWorkersPercent must be >=0 && <=1');
+        }
+    }
+
+    /**
+     * @param string $request
+     * @param string $domain
+     * @return bool
+     */
+    abstract protected function isRisky(string $request, string $domain): bool;
+
+    protected function randomRiskyWorker(int $riskyWorkers, int $totalWorkers): int
+    {
+        return rand($riskyWorkers, $totalWorkers - 1);
+    }
+
+    protected function randomSafeWorker(int $riskyWorkers): int
+    {
+        return rand(0, $riskyWorkers - 1);
+    }
+
+    protected function resolveWorkerId(Server $server, ?string $data): int
+    {
+        $totalWorkers = $server->setting['worker_num'];
+
+        // If data is not set, we can send the request to any worker.
+        // First we try to pick an idle worker, otherwise we randomly pick a worker.
+        if (empty($data)) {
+            for ($i = 0; $i < $totalWorkers; $i++) {
+                if ($server->getWorkerStatus($i) === SWOOLE_WORKER_IDLE) {
+                    return $i;
+                }
+            }
+            return rand(0, $totalWorkers - 1);
+        }
+
+        // Each worker has a numeric ID, starting from 0 and incrementing
+        // From 0 to $riskyWorkers, we consider safe workers
+        // From $riskyWorkers to $totalWorkers, we consider risky workers
+        $riskyWorkers = (int) floor($totalWorkers * $this->riskyWorkersPercent); // Absolute number of risky workers
+
+        $headers = explode("\n", strstr($data, "\r\n", true));
+        $request = $headers[0];
+        $domain = '';
+        if (count($headers) > 1) {
+            $domain = trim(explode('Host: ', $headers[1])[1]);
+        }
+
+        $risky = $this->isRisky($request, $domain);
+
+        if ($risky) {
+            // If risky request, only consider risky workers
+            for ($j = $riskyWorkers; $j < $totalWorkers; $j++) {
+                /** Reference https://openswoole.com/docs/modules/swoole-server-getWorkerStatus#description */
```

Comment:
i wouldn't reference openswoole, its an outdated fork and already quite behind 👍🏻 

rather use https://wiki.swoole.com/en/#/